### PR TITLE
Store access token in secureJsonData

### PR DESCRIPTION
**Pull request by nineinchnick**
Ref: https://grafana.com/developers/plugin-tools/how-to-guides/data-source-plugins/add-authentication-for-data-source-plugins#store-configuration-in-securejsondata

### CODE REVIEW

File: src/ConfigEditor.tsx [hunk 1]
```diff
@@ -14,8 +14,11 @@ export class ConfigEditor extends PureComponent<Props, State> {
       onOptionsChange({...options, jsonData: {...options.jsonData, enableImpersonation: event.target.checked}})
     }
     const onTokenChange = (event: ChangeEvent<HTMLInputElement>) => {
-      onOptionsChange({...options, jsonData: {...options.jsonData, accessToken: event.target.value}})
+      onOptionsChange({...options, secureJsonData: {...options.secureJsonData, accessToken: event.target.value}})
```
Comment (SemionPar) [inline src/ConfigEditor.tsx:17]: It is not clear to me what makes it actually secure? Is this just a name change?

Comment (nineinchnick) [reply src/ConfigEditor.tsx:17]: It's supposed to store the access token in a different structure - `secureJsonData` instead of `jsonData`.

Comment (nineinchnick) [reply src/ConfigEditor.tsx:17]: I've been following https://grafana.com/developers/plugin-tools/how-to-guides/data-source-plugins/add-authentication-for-data-source-plugins#store-configuration-in-securejsondata

Comment (SemionPar) [reply src/ConfigEditor.tsx:17]: I see - now it makes sense!

File: src/ConfigEditor.tsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import React, { ChangeEvent, PureComponent } from 'react';
 import { DataSourceHttpSettings, InlineField, InlineSwitch, Input } from '@grafana/ui';
```
Comment (jschroth) [inline src/ConfigEditor.tsx:2]:
```suggestion
import { DataSourceHttpSettings, InlineField, InlineSwitch, SecretInput } from '@grafana/ui';
```

File: src/ConfigEditor.tsx [hunk 3]
```diff
@@ -41,16 +44,26 @@ export class ConfigEditor extends PureComponent<Props, State> {
...
             </InlineField>
+            <InlineField
+                label="Reset access token"
+                tooltip="Clear any previously set access token"
+                labelWidth={26}
+            >
+              <InlineSwitch
+                  onChange={onResetToken}
+              />
+            </InlineField>
```
Comment (jschroth) [inline src/ConfigEditor.tsx:66]:
```suggestion
                <SecretInput
                  value={options.secureJsonData?.accessToken ?? ''}
                  isConfigured={options.secureJsonFields?.accessToken}
                  onChange={onTokenChange}
                  width={40}
                  onReset={onResetToken}
                />
            </InlineField>
```

**Review by jschroth — COMMENTED** [review]
Thanks for starting this. I tested this with these recommended changes here and it seems to work. The only thing missing is using the secure data:

in `settings.go` add something like the following before the return statement

```
	if token, exists := config.DecryptedSecureJSONData["accessToken"]; exists {
		s.AccessToken = token
	}
```

File: src/ConfigEditor.tsx (hunk 2 above)
Comment (nineinchnick) [reply src/ConfigEditor.tsx]: Applied, thanks!

**Review by jschroth — APPROVED** [review]
Not sure that my approval matters, but this works perfectly!

**Comment by nineinchnick** [comment]
It does, thank you for testing this!

**Comment by llamafilm** [comment]
Is there anything we can do to help accelerate releasing this?

**Comment by nineinchnick** [comment]
You can put pressure on me. Just kidding, but asking for it is enough. As usual, I can't promise anything, but I'll try releasing a new version this week.

**Comment by nineinchnick** [comment]
I haven't gotten to it yet. The biggest stopper for me is that I have to test the changes manually, and I do this so infrequently, that I lose time remembering how to run a particular version of Grafana locally.
It would speed things up significantly if we had some e2e test workflow, running both Trino and Grafana. I wouldn't have to do manual testing, and it would serve as a reference on how to run things locally, if needed.

**Comment by jschroth** [comment]
@nineinchnick I tested the changes from my end and it worked with my setup; although, I was using a production trino instance so it was not something had to setup manually. The grafana piece was just running the plugin locally in a docker container. I wrote up these notes to remind me of what I need to do get the plugin running in docker and then connect to `http://localhost:3000`.

```
# get plugin source
git clone https://github.com/trinodb/grafana-trino.git
cd grafana-trino
git fetch origin pull/277/head:pr277
git checkout pr277

# node + build
fnm use 19
yarn install
yarn dev

# mage (run `go mod tidy` if needed)
mage -v

# run
docker run --rm -p 3000:3000 \
  -v "$(pwd):/var/lib/grafana/plugins/trino" \
  -e "GF_PLUGINS_ALLOW_LOADING_UNSIGNED_PLUGINS=trino-datasource" \
  --name=grafana  grafana/grafana-enterprise
```

**Comment by nineinchnick** [comment]
I added E2E tests, released v1.0.10 and submitted it to Grafana for review. Hopefully it'll only take a few days, but I wouldn't be surprised if it'll get reviewed after the new year.

**Comment by nineinchnick** [comment]
1.0.10 has been published! Thanks everyone for your patience: https://grafana.com/grafana/plugins/trino-datasource/

**Pull request merged**